src/idl_gen_cpp.cpp

```diff
@@ -2273,11 +2273,20 @@ class CppGenerator : public BaseGenerator {
   void GenKeyFieldMethods(const FieldDef &field) {
     FLATBUFFERS_ASSERT(field.key);
     const bool is_string = (IsString(field.value.type));
+    const bool is_array = (IsArray(field.value.type));
 
     code_ += "  bool KeyCompareLessThan(const {{STRUCT_NAME}} *o) const {";
     if (is_string) {
       // use operator< of flatbuffers::String
       code_ += "    return *{{FIELD_NAME}}() < *o->{{FIELD_NAME}}();";
+    } else if (is_array){
+      const auto &elem_type = field.value.type.VectorType();
+      if (IsScalar(elem_type.base_type)) {
+        code_ += "    for (auto i = 0; i < {{FIELD_NAME}}()->size(); i++){ ";
+        code_ += "      if ({{FIELD_NAME}}()->Get(i) != o->{{FIELD_NAME}}()->Get(i)) return {{FIELD_NAME}}()->Get(i) < o->{{FIELD_NAME}}()->Get(i);";
```

Comment:
I would put the return part on a separate line. There is no real benefit to streamline everything in the generated code, and it makes reading the generator code easier.